staging/src/k8s.io/apimachinery/pkg/util/wait/wait_test.go

```diff
@@ -455,18 +455,46 @@ func TestWaitFor(t *testing.T) {
 	}
 }
 
-func TestWaitForWithDelay(t *testing.T) {
-	done := make(chan struct{})
-	defer close(done)
-	WaitFor(poller(time.Millisecond, ForeverTestTimeout), func() (bool, error) {
+func TestWaitForWithClosedChannel(t *testing.T) {
+	stopCh := make(chan struct{})
+	close(stopCh)
+	start := time.Now()
+	err := WaitFor(poller(ForeverTestTimeout, ForeverTestTimeout), func() (bool, error) {
+		return false, nil
+	}, stopCh)
+	duration := time.Now().Sub(start)
+	// The WaitFor should returns immediately. So the duration is closed to 0s.
+	// This condition ensures that if the WaitFor returns error caused by poller rather
+	// than stopCh, it will trigger an error.
```

Comment:
Sorry, I don't follow this comment. Are you trying to explain why you did the check in line 472?